workos/utils/_base_http_client.py

```diff
@@ -0,0 +1,191 @@
+                response, error=error, error_description=error_description
+            )
+        elif status_code >= 500 and status_code < 600:
+            raise ServerException(response)
+
+    def _prepare_request(
+        self,
+        path: str,
+        method: str = REQUEST_METHOD_GET,
+        params: dict = None,
+        headers: dict = None,
+        token: str = None,
+    ) -> PreparedRequest:
+        """Executes a request against the WorkOS API.
+
+        Args:
+            path (str): Path for the api request that'd be appended to the base API URL
+
+        Kwargs:
+            method (str): One of the supported methods as defined by the REQUEST_METHOD_X constants
+            params (dict): Query params or body payload to be added to the request
+            token (str): Bearer token
+
+        Returns:
+            dict: Response from WorkOS
+        """
+        url = self._generate_api_url(path)
+        headers = self._build_headers(headers, token)
+        bodyless_http_method = method.lower() in [
+            REQUEST_METHOD_DELETE,
+            REQUEST_METHOD_GET,
+        ]
+
+        # Remove any parameters that are None
+        if params is not None:
+            params = (
+                {k: v for k, v in params.items() if v is not None}
+                if bodyless_http_method
+                else params
+            )
+
+        # We'll spread these return values onto the HTTP client request method
+        if bodyless_http_method:
+            return {
+                "method": method,
+                "url": url,
+                "headers": headers,
+                "params": params,
+                "timeout": self.timeout,
+            }
+        else:
+            return {
+                "method": method,
+                "url": url,
+                "headers": headers,
+                "json": params,
+                "timeout": self.timeout,
+            }
+
+    def _handle_response(self, response: httpx.Response) -> dict:
+        response_json = None
+        content_type = (
+            response.headers.get("content-type")
+            if response.headers is not None
+            else None
+        )
+        if content_type is not None and "application/json" in content_type:
+            try:
+                response_json = response.json()
+            except ValueError:
+                raise ServerException(response)
+
+        self._maybe_raise_error_by_status_code(response, response_json)
+
+        return response_json
+
+    @property
+    def base_url(self) -> str:
+        return self._base_url
+
+    @base_url.setter
+    def base_url(self, url: str) -> None:
+        """Creates an accessible template for constructing the URL for an API request.
+
+        Args:
+            base_api_url (str): Base URL for api requests
+        """
+        self._base_url = "{}{{}}".format(self._enforce_trailing_slash(url))
+
+    @property
+    def default_headers(self) -> "dict[str, str]":
```

Comment:
When do you `"dict[str, str]"` in parseltongue and when do you `Dict[str, str]`?

Reply:
I think this is Python versioning differences. I'll change to Dict[str, str] to match our other type hints, but in newer versions of Python there is no difference it's just a quality of life improvement (no import).